src/Core/Batch/BatchRunner.php

```
@@ -0,0 +1,183 @@
+     * @param JobSubmitInterface $jobSubmitter [optional] The JobSubmitter
+     *        object to use. **Defaults to** null.
+     */
+    public function __construct(
+        ConfigStorageInterface $configStorage = null,
+        JobSubmitInterface $jobSubmitter = null
+    ) {
+        if ($configStorage === null || $jobSubmitter === null) {
+            if ($this->isSysvIPCLoaded() && $this->isDaemonRunning()) {
+                $configStorage = new SysvConfigStorage();
+                $jobSubmitter = new SysvJobSubmitter();
+            } else {
+                $configStorage = InMemoryConfigStorage::getInstance();
+                $jobSubmitter = $configStorage;
+            }
+        }
+        $this->configStorage = $configStorage;
+        $this->submitter = $jobSubmitter;
+        $this->loadConfig();
+    }
+
+    /**
+     * Register a job for batch execution.
+     *
+     * @param string $identifier Unique identifier of the job.
+     * @param callable $func Any Callable except for Closure. The callable
+     *        should accept an array of items as the first argument.
+     * @param array $options [optional] {
+     *     Configuration options.
+     *
+     *     @type int $batchSize The size of the batch.
+     *     @type int $callPeriod The period in seconds from the last execution
+     *               to force executing the job.
+     *     @type int $workerNum The number of child processes. It only takes
+     *               effect with the {@see \Google\Cloud\Core\BatchDaemon}.
+     *     @type string $bootstrapFile A file to load before executing the
+     *                  job. It's needed for registering global functions.
+     * }
+     * @return boolean true on success, false on failure
+     */
+    public function registerJob($identifier, $func, $options = [])
+    {
+        if ($func instanceof \Closure) {
+            trigger_error('Closure is not allowed', E_USER_NOTICE);
```

I'm not sure that a E_USER_NOTICE is high enough. Lots of PHP configurations don't display notices by default (ugh), so an invalid configuration could silently fail. Perhaps E_USER_ERROR or a \InvalidArgumentException would be better.